pkg/component/gardener/dashboard/rbac.go

```diff
@@ -0,0 +1,170 @@
+			Labels: GetLabels(),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{authenticationv1.GroupName},
+				Resources: []string{"tokenreviews"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups: []string{gardencorev1beta1.GroupName},
+				Resources: []string{"quotas", "projects", "shoots", "controllerregistrations"},
+				Verbs:     []string{"list", "watch"},
+			},
+			{
+				APIGroups: []string{apiregistrationv1beta1.GroupName},
+				Resources: []string{"apiservices"},
+				Verbs:     []string{"get"},
+			},
+			{
+				APIGroups:     []string{corev1.GroupName},
+				Resources:     []string{"configmaps"},
+				Verbs:         []string{"get"},
+				ResourceNames: []string{v1beta1constants.ClusterIdentity},
+			},
+			{
+				APIGroups: []string{corev1.GroupName},
+				Resources: []string{"resourcequotas"},
+				Verbs:     []string{"list", "watch"},
+			},
+			// TODO: Remove once https://github.com/gardener/monitoring/issues/11 is resolved.
+			{
+				APIGroups: []string{corev1.GroupName},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"get"},
+			},
+		},
+	}
+}
+
+func (g *gardenerDashboard) clusterRoleBinding(serviceAccountName string) *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleBindingName,
+			Labels: GetLabels(),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     clusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      serviceAccountName,
+			Namespace: metav1.NamespaceSystem,
+		}},
+	}
+}
+
+func (g *gardenerDashboard) clusterRoleBindingTerminal() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleBindingTerminalName,
+			Labels: GetLabels(),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     v1beta1constants.ClusterRoleNameGardenerAdministrators,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      serviceAccountNameTerminal,
+			Namespace: metav1.NamespaceSystem,
+		}},
+	}
+}
+
+func (g *gardenerDashboard) clusterRoleTerminalProjectMember() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleTerminalProjectMemberName,
+			Labels: utils.MergeStringMaps(GetLabels(), map[string]string{v1beta1constants.LabelKeyAggregateToProjectMember: "true"}),
+		},
+		Rules: []rbacv1.PolicyRule{{
+			APIGroups: []string{"dashboard.gardener.cloud"},
+			Resources: []string{"terminals"},
+			Verbs:     []string{"create", "delete", "deletecollection", "get", "list", "watch", "patch", "update"},
+		}},
+	}
+}
+
+func (g *gardenerDashboard) role() *rbacv1.Role {
+	return &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gitHubWebhookRoleName,
+			Namespace: metav1.NamespaceSystem,
```

Comment:
https://github.com/gardener/dashboard/blob/69c3963dc80532c704591ce82be1e9835dbe5ad6/backend/lib/github/webhook/handler.js#L18

currently the dashboard tries to read/write the leases in `namespace = process.env.POD_NAMESPACE || 'garden'`

that's why it currently fails with 
```
2024-04-19 15:14:59.521 res  : HTTP/2 403 Forbidden [120] GET /apis/coordination.k8s.io/v1/namespaces/garden/leases?fieldSelector=metadata.name%3Dgardener-dashboard-github-webhook&resourceVersion=0&limit=500 9ms
2024-04-19 15:14:59.521 error: Failed to call paginated list coordination.k8s.io/v1, Kind=Lease: leases.coordination.k8s.io "gardener-dashboard-github-webhook" is forbidden: User "system:serviceaccount:kube-system:gardener-dashboard" cannot list resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
```